### PR TITLE
Sweets/smart wallet

### PR DESCRIPTION
**Pull request by sweetmantech**

### CODE REVIEW

File: components/ProfilePage/SmartWalletContents.tsx [hunk 1]
```diff
@@ -4,14 +4,28 @@ import { useProfileProvider } from "../../providers/ProfileContext"
...
   const { cre8orNumber } = useProfileProvider()
   const [ownedNfts, setOwnedNfts] = useState([])
-
+  const [hasSmartWallet, setHasSmartWallet] = useState(true)
+  const provider = getDefaultProvider(process.env.NEXT_PUBLIC_TESTNET ? 5 : 1)
   useEffect(() => {
     const init = async () => {
-      const smartWalletAddress = await getSmartWallet(cre8orNumber)
+      const smartWalletAddress = await getSmartWallet(85)
+      //
```
Comment (towardsnewtech) [inline components/ProfilePage/SmartWalletContents.tsx:18]: Do we absolutely have to use 85 to get the address of smart wallet? ❔
It feels like it's hardcoded somewhere. 💭
If so, please ignore this comment.

**Review by towardsnewtech — COMMENTED** [review]
LGTM! 👍

**Review by towardsnewtech — APPROVED** [review]

File: components/ProfilePage/SmartWalletContents.tsx (hunk 1 above)
Comment (crazzywizard) [reply components/ProfilePage/SmartWalletContents.tsx]: I agree with ziad. If it is necessary to use a magic number, better to set it as a variable with a clear meaning variable name.

File: components/ProfilePage/Deploy6551AndMintButton.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import abi from "../../lib/abi-dna-minter.json"
+import { useProfileProvider } from "../../providers/ProfileContext"
+import { useEthersSigner } from "../../hooks/useEthersSigner"
+
+const Deploy6551AndMintDNAButton = () => {
+  const { cre8orNumber } = useProfileProvider()
+  const signer = useEthersSigner({ chainId: process.env.NEXT_PUBLIC_TESTNET ? 5 : 1 })
+
+  const onClick = async () => {
+    const dnaMinter = "0xeBBe9F31ef46ae687a60eB25F38eF8B8b2311E25"
```
Comment (crazzywizard) [inline components/ProfilePage/Deploy6551AndMintButton.tsx]: is this supposed to be hardcoded contract address? We usually set them in env variables.

File: components/ProfilePage/SmartWalletContents.tsx [hunk 3]
```diff
@@ -4,14 +4,28 @@ import { useProfileProvider } from "../../providers/ProfileContext"
...
+      // Fetch the contract code
+      const code = await provider.getCode(smartWalletAddress)
+
+      if (code === "0x") {
+        console.log(`No contract deployed at address: ${smartWalletAddress}`)
```
Comment (crazzywizard) [inline components/ProfilePage/SmartWalletContents.tsx]: lint will complain about this.

Comment (crazzywizard) [reply components/ProfilePage/SmartWalletContents.tsx]: my guess is a lot of these hard coded values will be replaced after testing?

**Review by crazzywizard — CHANGES_REQUESTED** [review]
requesting changes to bring to attention about console logs and hard coded values.

File: components/ProfilePage/SmartWalletContents.tsx (hunk 1 above)
Comment (sweetmantech) [reply components/ProfilePage/SmartWalletContents.tsx]: my bad, it was late
updating now

Comment (sweetmantech) [reply components/ProfilePage/SmartWalletContents.tsx]: updated from `85` => `cre8orNumber`

Screenshot 2023-08-09 at 8 11 59 AM

File: components/ProfilePage/Deploy6551AndMintButton.tsx (hunk 2 above)
Comment (sweetmantech) [reply components/ProfilePage/Deploy6551AndMintButton.tsx]: good catch, setting this to `NEXT_PUBLIC_DNA_MINTER` 👍

File: components/ProfilePage/SmartWalletContents.tsx (hunk 3 above)
Comment (sweetmantech) [reply components/ProfilePage/SmartWalletContents.tsx]: removed dev logging ✅

**Pull request merged**